src/cryoemservices/util/clem_array_functions.py

```diff
@@ -561,6 +564,205 @@ def stretch_image_contrast(
     return arr_new
 
 
+def align_image_to_self(
+    array: np.ndarray,
+    start_from: Literal["beginning", "middle", "end"] = "beginning",
+) -> np.ndarray:
+    """
+    Use PyStackReg to correct for drift in an image stack.
+    """
+    # Record initial dtype and intensities
+    dtype = str(array.dtype)
+    vmin: int | float = array.min()
+    vmax: int | float = array.max()
+
+    # Check if an image or a stack has been passed to the function
+    shape = array.shape
+    if len(shape) <= 2 or (
+        len(shape) == 3 and shape[-1] in (3, 4)  # Check for 2D RGB or RGBA images
```

Comment:
This would unintentionally exclude stacks with 3 or 4 images. Are those possible?